dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/metadata/SIARDDK128MetadataImportStrategy.java

```diff
@@ -0,0 +1,472 @@
+
+    try {
+      xsdSchema = schemaFactory.newSchema(new StreamSource(xsdInputStream));
+    } catch (SAXException e) {
+      throw new ModuleException()
+              .withMessage("Error reading metadata XSD file: " + pathStrategy.getXsdFilePath(SIARDDKConstants.DOC_INDEX))
+              .withCause(e);
+    }
+    InputStream inputStreamXml = null;
+    Unmarshaller unmarshaller;
+    try {
+      unmarshaller = context.createUnmarshaller();
+      unmarshaller.setSchema(xsdSchema);
+      inputStreamXml = new FileInputStream(pathStrategy.getMainFolder().getPath().toString() + "/" +
+              pathStrategy.getXmlFilePath(SIARDDKConstants.DOC_INDEX));
+      JAXBElement<DocIndexType> jaxbElement = (JAXBElement<DocIndexType>) unmarshaller.unmarshal(inputStreamXml);
+      return jaxbElement.getValue();
+    } catch (JAXBException e) {
+      throw new ModuleException().withMessage("Error while Unmarshalling JAXB").withCause(e);
+    } finally {
+      try {
+        xsdInputStream.close();
+        if (inputStreamXml != null) {
+          inputStreamXml.close();
+          xsdInputStream.close();
+        }
+      } catch (IOException e) {
+        logger.debug("Could not close xsdStream", e);
+      }
+    }
+  }
+
+  private List<ForeignKey> getVirtualForeignKeys(ColumnsType columns, String tableId) {
+    List<ForeignKey> virtualForeignKeys = new ArrayList<>();
+    for (ColumnType column : columns.getColumn()) {
+      if (column.getFunctionalDescription() != null && column.getFunctionalDescription().contains(FunctionalDescriptionType.DOKUMENTIDENTIFIKATION)) {
+        ForeignKey virtualForeignKey = new ForeignKey();
+        virtualForeignKey.setReferencedSchema(getImportAsSchemaName());
+        virtualForeignKey.setName("FK_virtual_table");
+        virtualForeignKey.setReferencedTable("virtual_table");
+        Reference reference = new Reference(column.getName(), "dID");
+        List<Reference> referenceList = new ArrayList<>();
+        referenceList.add(reference);
+        virtualForeignKey.setReferences(referenceList);
+        virtualForeignKey.setId(String.format("%s.%s", tableId, "FK_virtual_table"));
+        virtualForeignKeys.add(virtualForeignKey);
+      }
+    }
+
+    return virtualForeignKeys;
+  }
+
+  private TableStructure createVirtualTable() throws ModuleException {
+    try {
+      DocIndexType docIndexType = loadVirtualTableMetadata();
+      TableStructure virtualTable = new TableStructure();
+      virtualTable.setIndex(currentTableIndex++);
+      virtualTable.setSchema(getImportAsSchemaName());
+      virtualTable.setId(String.format("%s.%s", virtualTable.getSchema(), "virtual_table"));
+      virtualTable.setName("virtual_table");
+      virtualTable.setDescription("A virtual table");
+      virtualTable.setRows(docIndexType.getDoc().size());
+      virtualTable.setColumns(createVirtualTableColumns());
+      virtualTable.setPrimaryKey(createVirtualPrimaryKey("dID"));
```

Comment:
Do the same for columns, primary keys, foreign keys and anything else that is virtual.

Reply:
done